### ISSUE TITLE
atom.packages.resolvePackagePath(name) works incorrectly when directory with given name exists in current working directory

### ISSUE THREAD

**Issue by dmytrokyrychuk** [atom/atom#10982]
### Prerequisites
- [x] Can you reproduce the problem in [safe mode](https://atom.io/docs/latest/hacking-atom-debugging#check-if-the-problem-shows-up-in-safe-mode)?
- [x] Are you running the [latest version of Atom](https://atom.io/docs/latest/hacking-atom-debugging#update-to-the-latest-version)?
- [x] Did you check the [debugging guide](https://atom.io/docs/latest/hacking-atom-debugging)?
- [x] Did you check the [FAQs on Discuss](https://discuss.atom.io/c/faq)?
- [x] Are you reporting to the [correct repository](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#atom-and-packages)?
- [x] Did you [perform a cursory search](https://github.com/issues?q=is%3Aissue+user%3Aatom+-repo%3Aatom%2Felectron) to see if your bug or enhancement is already reported?

For more information on how to write a good [bug report](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#how-do-i-submit-a-good-bug-report) or [enhancement request](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#how-do-i-submit-a-good-enhancement-suggestion), see the `CONTRIBUTING` guide.
### Description

atom.packages.resolvePackagePath(name) works incorrectly when directory with given name exists in current working directory. 
### Steps to Reproduce

Run the following commands in Developer Tools Console.

``` js
> packageName = atom.packages.getAvailablePackageNames()[42]
< "autocomplete-html"
> atom.packages.resolvePackagePath(packageName)
< "/usr/share/atom/resources/app.asar/node_modules/autocomplete-html"
> require('fs').mkdir(packageName)
> atom.packages.resolvePackagePath(packageName)
< "autocomplete-html"
```

**Expected behavior:** Resolved path must be an actual package path.

**Actual behavior:** Resolved path may be a relative path to existing directory.
### Versions

```
$ atom --version
1.5.3
```

```
$ apm --version
apm  1.6.0
npm  2.13.3
node 0.10.40
python 2.7.6
git 1.9.1
```
### Proposal

[This line](https://github.com/atom/atom/blob/3b309792a531fb809d29c6742de956f50b3acec5/src/package-manager.coffee#L173) may be changed to the following:

``` coffee
return name if name !== path.basename(name) and fs.isDirectorySync(name)
```

**Comment by lee-dohm** [comment]
What platform is this on? I was unable to reproduce the problem using Atom v1.5.3 on Mac OS X 10.11.3. When I executed the `require('fs').mkdir(packageName)` command, I got an EACCESS error because the current working directory was `/`.

**Comment by orgkhnargh** [comment]
It's on Linux (Xubuntu 14.04). I installed Atom from deb package. Current working directory is the user's home directory, so I do not get an EACCESS error.

**Comment by lee-dohm** [comment]
Thanks @orgkhnargh!

**Comment by amytruong** [comment]
I was able to repro this @lee-dohm :+1:

**Comment by lee-dohm** [comment]
Awesome, thanks @amytruong :grinning:

**Comment by hedefalk** [comment]
+1

```
packageName = atom.packages.resolvePackagePath(atom.packages.getAvailablePackageNames()[17])
"/Users/viktor/.atom/packages"
packageName = atom.packages.resolvePackagePath(atom.packages.getAvailablePackageNames()[18])
"/Users/viktor/.atom/packages/ensime_update_coursier"
```

There is no "ensime_update_coursier" package, it's just a folder inside my 'Ensime' package…

**Comment by CptFox** [comment]
A year later this still seems to be an issue using Atom Beta (1.17.0-b4) on Ubuntu 16.04 LTS. Any plans on fixing it ?

**Comment by orgkhnargh** [comment]
Just to clarify, `require('fs').mkdir(packageName)` is a quick way to reproduce this issue. Some directories may already exist on user's computer that interfere with certain packages names (e.g., `build`).

In my opinion, the `resolvePackagePath` method should only return a path if it belongs to `atom.packages.getPackageDirPaths()`. `if fs.isDirectorySync(name)` is not enough to validate that.

**Comment by lee-dohm** [comment]
@CptFox The awesome thing about Atom being open source is that everyone is free to contribute! If this is causing problems for you, perhaps you have some free time to submit a PR? I would be happy to review it for you.

**Comment by sadick254** [comment]
Thanks for reporting this!

This has already been fixed on the latest version of atom `v1.51.0`. Consider upgrading to the latest version.
